src/comparison/ugbio_comparison/vcf_pipeline_utils.py

```diff
@@ -0,0 +1,927 @@
+        )
+        self.index_vcf(output_vcf)
+        # shutil.rmtree(tempdir)
+
+    def fix_vcf_format(self, output_prefix: str):
+        """Legacy function to fix the PS field format in the old GIAB truth sets. The function overwrites the input file
+
+        Parameters
+        ----------
+        output_prefix : str
+            Prefix of the input and the output file (without the .vcf.gz)
+        """
+        self.__execute(f"gunzip -f {output_prefix}.vcf.gz")
+        with open(f"{output_prefix}.vcf", encoding="utf-8") as input_file_handle:
+            with open(f"{output_prefix}.tmp", "w", encoding="utf-8") as output_file_handle:
+                for line in input_file_handle:
+                    if line.startswith("##FORMAT=<ID=PS"):
+                        output_file_handle.write(line.replace("Type=Integer", "Type=String"))
+                    else:
+                        output_file_handle.write(line)
+        self.__execute(f"mv {output_file_handle.name} {input_file_handle.name}")
+        self.__execute(f"bgzip {input_file_handle.name}")
+        self.index_vcf(f"{input_file_handle.name}.gz")
+
+    def annotate_tandem_repeats(self, input_file: str, reference_fasta: str) -> str:
+        """Runs VariantAnnotator on the input file to add tandem repeat annotations (maybe others)
+
+        Parameters
+        ----------
+        input_file : str
+            vcf.gz file
+        reference_fasta : str
+            Reference file (should have .dict file nearby)
+
+        Creates a copy of the input_file with .annotated.vcf.gz and the index
+        Returns
+        -------
+        path to output file: str
+        """
+
+        output_file = input_file.replace("vcf.gz", "annotated.vcf.gz")
+        self.__execute(f"gatk VariantAnnotator -V {input_file} -O {output_file} -R {reference_fasta} -A TandemRepeat")
+        return output_file
+
+    def transform_hom_calls_to_het_calls(self, input_file_calls: str, output_file_calls: str) -> None:
+        """Reverse homozygous reference calls in deepVariant to filtered heterozygous so that max recall can be
+        calculated
+
+        Parameters
+        ----------
+        input_file_calls : str
+            Input file name
+        output_file_calls : str
+            Output file name
+        """
+
+        with pysam.VariantFile(input_file_calls) as input_file:
+            with pysam.VariantFile(output_file_calls, "w", header=input_file.header) as output_file:
+                for rec in input_file:
+                    if (
+                        rec.samples[0]["GT"] == (0, 0)
+                        or rec.samples[0]["GT"] == (None, None)
+                        and "PASS" not in rec.filter
+                    ):
+                        rec.samples[0]["GT"] = (0, 1)
+                    output_file.write(rec)
+        self.index_vcf(output_file_calls)
+
+
+def _fix_errors(df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
```suggestion
def fix_errors(df: pd.DataFrame) -> pd.DataFrame:
```

Even consider a moer descriptive name for this function

Reply:
why? this is an internal function?